PlutoHelperAgent/SharedFunctions.m

```diff
@@ -0,0 +1,236 @@
+    
+    NSArray *failedReturnStrings = [NSArray arrayWithObjects:
+                                    @"Username not loaded",
+                                    @"Password not loaded",
+                                    nil];
+    
+    return failedReturnStrings;
+    
+}
+
++ (NSString *) login_to_project_server {
+    
+    NSArray *dataFromKeychain = [self load_data_from_keychain];
+    
+    NSString *URLToUse = [NSString stringWithFormat: @"%@/api/login", [[NSUserDefaults standardUserDefaults] stringForKey:@"project_locker_url"]];
+    
+    NSURL *url = [NSURL URLWithString:URLToUse];
+    NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:config];
+    
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:url];
+    request.HTTPMethod = @"POST";
+    [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    NSLog(@"Request %@", request);
+    
+    NSDictionary *dictionary = @{@"username": dataFromKeychain[0], @"password": dataFromKeychain[1]};
+    NSError *error = nil;
+    NSData *data = [NSJSONSerialization dataWithJSONObject:dictionary
+                                                   options:kNilOptions error:&error];
+    
+    if (!error) {
+        NSURLSessionUploadTask *uploadTask = [session uploadTaskWithRequest:request
+                                                                   fromData:data completionHandler:^(NSData *data,NSURLResponse *response,NSError *error) {
+                                                                       NSString *datastring = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+                                                                       NSLog(@"Data %@", datastring);
+                                                                       NSLog(@"Response %@", response);
+                                                                       if (error != NULL) {
+                                                                           NSLog(@"Error %@", error);
+                                                                           
+                                                                       }
+                                                                       
+                                                                       
+                                                                   }];
+        
+        [uploadTask resume];
+    }
+    
+    sleep(1);
+    
+    NSHTTPCookie *cookie;
+    NSHTTPCookieStorage *cookieJar = [NSHTTPCookieStorage sharedHTTPCookieStorage];
+    for (cookie in [cookieJar cookies]) {
+        NSLog(@"%@", cookie);
+    }
+    
+    NSString *URLToUse2 = [NSString stringWithFormat: @"%@/api/isLoggedIn", [[NSUserDefaults standardUserDefaults] stringForKey:@"project_locker_url"]];
```

Comment:
seperate this lot into a "check_logged_in" method that can be called seperately

Reply:
Done